core/src/main/java/org/apache/accumulo/core/clientImpl/TableOperationsImpl.java

```diff
@@ -1035,6 +1031,30 @@ public void modifyProperties(String tableName, final Consumer<Map<String,String>
     }
   }
 
+  @Override
+  public void modifyProperties(String tableName, final Consumer<Map<String,String>> mapMutator)
```

Comment:
Does it make sense to return a Future<Void> here instead? That would allow the caller to potentially perform other logic before checking the result and the caller has to handle the exceptions when calling Future.get().

Reply:
I would like to see Accumulo support async APIs.  I think that should be done with consideration of the entire API and looking at what APIs we want to add.  Not sure about doing that as a one off here.  Would need to create and manage a thread  or thread pool to do that here, which is another reason to consider it API wide maybe (maybe we want to allow passing in an executor for async APIs?)

I would also like to explore making the read and write pipelines in the tserver and sserver asynchronous.  I think this could support more client connections per server and would be good for making them  more responsive in the case we had discussed in the past of having many sservers and few tservers.  It think it would be good to make the server pipelines async at the same time as adding async APIs to Accumulo so that the two could influence each other.

Reply:
Async APIs and server pipelines may be a good 3.0 project.

Reply:
> Async APIs and server pipelines may be a good 3.0 project.

I agree with this, I think adding Async APIs for the next major version would be a nice addition.